### PR TITLE
Request for comments: deciding upon themes in ECL v3

### PR DESCRIPTION
**Pull request by kalinchernev**
I've tried to organize the information as concisely as possible.

Feel free to suggest changes with https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request#applying-a-suggested-change or pushing code directly.

### CODE REVIEW

File: docs/decisions/005-themes.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+This option has been discarded from the very beginning because it cannot achieve the goal without increased complexity and code duplication.
+
+#### Option 2: "themes" is single package, single preset generates multiple themes output
+
+This option has been discarded as too complex for the worth of the result. The complexity comes from SASS language limitations as dynamic imports are not supported. Every dynamic theme generation approach with SASS involves workarounds which were not acceptable for the team.
+
+#### Option 3: "themes" is single package, multiple presets generate multiple themes output
+
+- `@ecl/ecl-base` - common resources
```
Comment (emeryro) [inline docs/decisions/005-themes.md]:
```suggestion
- `@ecl/ecl-base` - common variables
```

File: docs/decisions/005-themes.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+- `@ecl/ecl-theme` - set of variables, one per theme
```
Comment (emeryro) [inline docs/decisions/005-themes.md]:
```suggestion
- `@ecl/ecl-theme` - set of variables related to the themes (global parameters), one set per theme
```

File: docs/decisions/005-themes.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Resources are: logo, favicons, icons and other similar types of assets which are not SCSS or JavaScript and are used by consumers of ECL.
+
+#### Option 1: separate package per system or per theme
+
+Current organization of v2.
+
+- `@ecl/ec-resources-logo`
+- `@ecl/eu-resources-logo`
+
+#### Option 2: single package
```
Comment (emeryro) [inline docs/decisions/005-themes.md:84]: That's what we had in the begining, but we split it in several packages to allow users to grab just what they need

File: docs/decisions/005-themes.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+- `@ecl/resources`
```
Comment (emeryro) [inline docs/decisions/005-themes.md:86]:
```suggestion
- `@ecl/resources`

#### Option 3: include resources in themes

This option does imply that each theme comes in a dedicates package (see ahead).
Resources useful to display a theme are stored directly in it.
To limit duplication, some global resources may still be put in dedicated packages.
```

File: docs/decisions/005-themes.md (hunk 3 above)
Comment (kalinchernev) [reply docs/decisions/005-themes.md:84]: Didn't know. Please do add this info similarly to https://github.com/ec-europa/europa-component-library/pull/1703/files#diff-25588a9df5ef0f08c630d3d1e730dffdR42

File: docs/decisions/005-themes.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+| **Driver**    | @kalinchernev                                         |
+| **Approver**  |                                                       |
+| **Consulted** | @emeryro                                              |
+| **Informed**  | [@team](https://github.com/orgs/ec-europa/teams/inno) |
+
+## Decisions
+
+## Context
+
+- ECL v2 is based on the concept of `systems` such as EC and EU
```
Comment (planctus) [inline docs/decisions/005-themes.md]:
```suggestion
- ECL v2 was made in order to generate two independent design systems, EC and EU.
```

File: docs/decisions/005-themes.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+- ECL v3 needs to accommodate the concept of `themes`.
```
Comment (planctus) [inline docs/decisions/005-themes.md]:
```suggestion
- ECL V3 starts from the assumption that there is only one source template and one design system with multiple `themes`.
```

**Comment by kalinchernev** [comment]
Thank you @planctus @emeryro for your contributions!

With the current summary and definitions, my personal preference would go for "Option 4: "themes" are multiple packages, multiple presets generate multiple themes output". It'll be straight-forward way to iterate on themes thinking in separation. If code becomes too similar or same it's easier to combine in 1 package later one rather than vice-versa.

I'd say resources can remain separate packages as in v2.

**Comment by emeryro** [comment]
I don't have a strong opinion here, but keeping things in separate packages is indeed the easiest way to go with the current toolchain (very close to how ECL 2 is handled).

**Review by planctus — APPROVED** [review]

**Review by emeryro — APPROVED** [review]

**Pull request merged**